### PR TITLE
Drop Python 2 support !

### PR DESCRIPTION
**Pull request by eduardo-rodrigues**
Towards a release 1.1.0 that ony supports Python >= 3.6 but is otherwise identical to 1.0.0.

### CODE REVIEW

**Comment by marinang** [comment]
What about adding the other scikit-hep packages, namely
* pyhf
* hepstats
* pylhe
* others ?

**Comment by eduardo-rodrigues** [comment]
It's already in "my pipeline" 👍 . It's just that I prefer to create well scoped PRs. Dropping Python 2 will make things easier ...

(As a matter of fact I had started to prepare a PR to add `hepstats` and remove the `modeling` submodule from here, since available in `hepstats`, and then realised that I should first drop Python 2 support to have a natural evolution of the repo :D.)

Thanks for the ping anyway.

**Comment by marinang** [comment]
@eduardo-rodrigues alright ;)

**Comment by eduardo-rodrigues** [comment]
BTW, `pylhe` isn't yet ready to be included. It really needs first a release with all recent (and important) developments. I'm pushing on that front ...

**Comment by eduardo-rodrigues** [comment]
Just in case I was not clear - I'm totally with you that we should include asap the important packages you list ;-).

**Comment by henryiii** [comment]
:clap:

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [2.7, 3.6, 3.8]
+        python-version: [3.6, 3.8]
```
Comment (henryiii) [inline .github/workflows/ci.yml:17]: Note to self for later - we need to add 3.9 here eventually. boost-histogram should support 3.9 in under a week, I think.

**Review by henryiii — APPROVED** [review]

**Comment by eduardo-rodrigues** [comment]
> 👏

I knew it that you're going to be happy :D.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (eduardo-rodrigues) [reply .github/workflows/ci.yml:17]: Indeed 3.9 is freshly out. Under a week sounds great.

**Pull request merged**